examples/multigpu_advtrain/run.py

```diff
@@ -0,0 +1,317 @@
+
+    with tf.variable_scope('preprocess'):
+        images_list = tf.split(images_batch, int(images_batch.shape[0]))
+        result_list = []
+        for img in images_list:
+            reshaped_img = tf.reshape(img, img.shape[1:])
+            processed_img = preproc_func(reshaped_img)
+            result_list.append(tf.expand_dims(processed_img, axis=0))
+        result_images = tf.concat(result_list, axis=0)
+    return result_images
+
+
+def model_train(sess, model, x_pre, x, y, predictions, X_train, Y_train,
+                predictions_adv=None, init_all=True, evaluate=None,
+                verbose=True, args=None, hparams=None):
+    """
+    Train a TF graph
+    :param sess: TF session to use when training the graph
+    :param x: input placeholder
+    :param y: output placeholder (for labels)
+    :param predictions: model output predictions
+    :param X_train: numpy array with training inputs
+    :param Y_train: numpy array with training outputs
+    :param hparams.save: boolean controlling the save operation
+    :param predictions_adv: if set with the adversarial example tensor,
+                            will run adversarial training
+    :param init_all: (boolean) If set to true, all TF variables in the session
+                     are (re)initialized, otherwise only previously
+                     uninitialized variables are initialized before training.
+    :param evaluate: function that is run after each training iteration
+                     (typically to display the test/validation accuracy).
+    :param verbose: (boolean) all print statements disabled when set to False.
+    :param args: dict or argparse `Namespace` object.
+                 Should contain `nb_epochs`, `learning_rate`,
+                 `batch_size`
+                 If hparams.save is True, should also contain 'train_dir'
+                 and 'filename'
+    :return: True if model trained
+    """
+    args = _FlagsWrapper(args or {})
+
+    # Check that necessary arguments were given (see doc above)
+    assert args.nb_epochs, "Number of epochs was not given in args dict"
+    assert args.learning_rate, "Learning rate was not given in args dict"
+    assert args.batch_size, "Batch size was not given in args dict"
+
+    if hparams.save:
+        assert args.train_dir, "Directory for save was not given in args dict"
+        assert args.filename, "Filename for save was not given in args dict"
+
+    if hparams.model_type == 'resnet_tf':
+        build_train_op = model.build_cost
+    else:
+        build_train_op = model_loss
+
+    # Define loss
+    with tf.variable_scope('train_loss'):
+        loss = build_train_op(y, predictions)
+        if predictions_adv is not None:
+            if hparams.only_adv_train:
+                loss = build_train_op(y, predictions_adv)
+            else:
+                adv_loss = build_train_op(y, predictions_adv)
+                loss = (loss + adv_loss) / 2
+
+    loss = tf.Print(loss, [loss])
```

Comment:
I'd suggest removing this, or if you want to keep it, add a ', "loss"' argument so that users can tell what is being printed to STDOUT. If users just see numbers without context they won't know what they are.

Reply:
Fixed in 05716f1.